### ISSUE TITLE
Heartbeat event

### ISSUE THREAD

**Issue by inad9300** [Atmosphere/atmosphere-javascript#264]
Is there a way to listen to heartbeats from the client side? Something like `onHeartbeat()`... If not, would it be possible to add it?

**Comment by inad9300** [comment]
Alternatively, the ability to send a heartbeat ("X") from the client would also be useful for my case – does this happen to be possible?

**Comment by jfarcand** [comment]
@inad9300 Yes it is doable. Take a look at https://github.com/Atmosphere/atmosphere-javascript/blob/7e7788b8d96cb587b945ed06fa95a9700afeb8f2/modules/javascript/src/main/webapp/javascript/atmosphere.js#L2913-L2917. You can probably add the `onHeartbeat()` there. If you can try it and do a pull request I can integrate it. Thanks

**Comment by inad9300** [comment]
Based on my experiences with Atmosphere (both front and back ends), I have decided to move away from it in favor of [Spring WebSockets](https://docs.spring.io/spring-framework/docs/current/spring-framework-reference/web.html#websocket), so if nobody else seems to need this feature, lets not bloat anymore the framework.

I have, nevertheless, out of curiosity, had a look at the source code. In particular, I passed it through TypeScript's compiler, and found a number of improvements and potential bugs that I summarized in https://github.com/Atmosphere/atmosphere-javascript/pull/265.

If you expect people to contribute in the future, I recommend you add automated testing to the project, as well as type safety and modules. Otherwise, modifying 3400+ lines source file full of legacy code for a project used by thousands is just too risky.
